### ISSUE TITLE
Setting maxParallelThreads to -1 in Json configuration doesn't work

### ISSUE THREAD

**Issue by BrennanConroy** [xunit/xunit#1629]
Didn't make a repro project since its so simple, but I can if you really want.

Our `xunit.runner.json` is as follows:
```json
{
  "longRunningTestSeconds": 30,
  "diagnosticMessages": true,
  "maxParallelThreads": -1
}
```
However, when running tests the output displays "(parallel test collections = on, max threads = 2)"

Looking through the code, it looks like your json parsing is ignoring `-1` and not setting the value in `TestAssemblyConfiguration` and later when reading it will be null and default to `Environment.ProcessorCount`
https://github.com/xunit/xunit/blob/0337edb69bc9830132c96249ae52004eb9dc70b2/src/xunit.runner.utility/Configuration/ConfigReader_Json.cs#L58

**Comment by anurse** [comment]
Repro project: https://github.com/anurse/ReproXunitMaxConcurrency

1. `cd ReproXunitMaxConcurrency`
2. `dotnet xunit -fxversion 2.0.0` (needed the `-fxversion` because for some reason xunit was trying to find the version "2.0" instead of "2.0.0", might be a different bug ;))

Expected (checked by adding `-maxthreads unlimited` to the command line) 

```
... bunch of internal diagnostics ...
  Discovering: ReproXunitMaxConcurrency (method display = ClassAndMethod)
  Discovered:  ReproXunitMaxConcurrency (running 1 test case)
  Starting:    ReproXunitMaxConcurrency (parallel test collections = on, max threads = unlimited)
```

Actual (when using the JSON setting and NOT using `-maxthreads unlimited` on the command line)

```
... same internal diagnostics ...
  Discovering: ReproXunitMaxConcurrency (method display = ClassAndMethod)
  Discovered:  ReproXunitMaxConcurrency (running 1 test case)
  Starting:    ReproXunitMaxConcurrency (parallel test collections = on, max threads = 12)
```

Note the max threads is `12`

**Comment by BrennanConroy** [comment]
Looks like the same is true for `[assembly: CollectionBehavior(MaxParallelThreads = n)]`
It isn't being considered when running the above scenario

@bradwilson

**Comment by bradwilson** [comment]
`-maxthreads` overrides any other value.

**Comment by BrennanConroy** [comment]
>Actual (when using the JSON setting and NOT using -maxthreads unlimited on the command line)

**Comment by onovotny** [comment]
Closing as fixed by #1632
